editor/editor_settings.cpp

```diff
@@ -1493,6 +1497,15 @@ Ref<ShortCut> ED_SHORTCUT(const String &p_path, const String &p_name, uint32_t p
 		ie->set_metakey(bool(p_keycode & KEY_MASK_META));
 	}
 
+	if (!EditorSettings::get_singleton()) {
+		Ref<ShortCut> sc;
+		sc.instance();
+		sc->set_name(p_name);
+		sc->set_shortcut(ie);
+		sc->set_meta("original", ie);
+		return sc;
```

Comment:
Shouldn't it just return `NULL` at the start of the method? What's the use case for `ED_SHORTCUT` when EditorSettings does not exist yet?

Reply:
The use case is certain constructors, such as EditorFileDialog, which use `ED_SHORTCUT` without EditorSettings being instantiated first. I guess returning NULL is valid too, but decided to make it behave as if the shortcut isn't found.